### PR TITLE
add globals when running an eval from cli, fixes #592

### LINKED ISSUE #592: `6to5-node -e` evaluates in an unsatisfactory context (e.g. no require)
```
> 6to5-node -e "console.log(require)"

/usr/local/lib/node_modules/6to5/bin/_6to5-node:49
  return vm.runInThisContext(code, filename);
            ^
ReferenceError: require is not defined
    at eval:1:13
    at _eval (/usr/local/lib/node_modules/6to5/bin/_6to5-node:49:13)
    at Object.<anonymous> (/usr/local/lib/node_modules/6to5/bin/_6to5-node:53:16)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:929:3
```

While in `node`:

```
> node -e "console.log(require)"
{ [Function: require]
  resolve: [Function],
  main: undefined,
  extensions: { '.js': [Function], '.json': [Function], '.node': [Function] },
  registerExtension: [Function],
  cache: {} }
```

### PR DESCRIPTION
**Pull request by bmeck**

### CODE REVIEW

**Comment by sebmck** [comment]
The only problem with this is that `require` is going to resolve relative to `bin` instead of the cwd. A more reliable solution would be to construct a new Module via the undocumented internal `module`.

**Comment by bmeck** [comment]
I was just matching what node does, it performs a wrapping step and does not create a Module. I'll patch this to resolve against the right dir.

**Comment by sebmck** [comment]
@bmeck Great! Should be as simple as using `require` and `exports` on the constructed module.

File: bin/_6to5-node [hunk 1]
```diff
@@ -56,7 +56,15 @@ if (program.eval || program.print) {
   var code = program.eval;
   if (!code || code === true) code = program.print;
 
-  var result = _eval(code, "eval");
+  global.__filename = "[eval]";
+  global.__dirname = process.cwd();
+  global.require = require;
```
Comment (eventualbuddha) [inline bin/_6to5-node]: Per the discussion with @sebmck, this should be something like this, right?

```js
global.require = function (path) {
  return require(pathJoin(process.cwd(), path));
};
```

Comment (sebmck) [inline bin/_6to5-node]: You'd just use the `module.require` that's built on line 64.

```javascript
global.require = module.require.bind(module);
```

**Pull request closed without merging**